### PR TITLE
Syncing psql e2e integration tests.

### PR DESCRIPTION
**Pull request by Vizerai**

### CODE REVIEW

**Comment by voulg** [comment]
ci/e2e-psql-v{11,12,13} seem to fail, is this expected?

**Comment by Vizerai** [comment]
They added pg-catalog to the backend which changed the expected results.  Had to fix a bunch of small formatting issues as well.

**Review by voulg — APPROVED** [review]
LGTM, but wait for Michael's review as well.

File: .ci/run-with-credentials.sh [hunk 1]
```diff
@@ -63,9 +63,37 @@ docker-build)
...
   docker push "${DOCKER_HOSTNAME}/${GOOGLE_CLOUD_PROJECT}/${DOCKER_REPOSITORY}/${DOCKER_IMAGE}"
   ;;
+e2e-psql)
+  PSQL_VERSION="$2"
+  GOOGLE_CLOUD_DATABASE_WITH_VERSION="${GOOGLE_CLOUD_DATABASE}_v${PSQL_VERSION}"
+#  create testing database
+  gcloud config set api_endpoint_overrides/spanner "https://${GOOGLE_CLOUD_ENDPOINT}/"
+  gcloud alpha spanner databases create "${GOOGLE_CLOUD_DATABASE_WITH_VERSION}" --instance="${GOOGLE_CLOUD_INSTANCE}" --database-dialect=POSTGRESQL
+  gcloud spanner databases ddl update "${GOOGLE_CLOUD_DATABASE_WITH_VERSION}" --instance="${GOOGLE_CLOUD_INSTANCE}" --ddl='CREATE TABLE users (id bigint PRIMARY KEY, age bigint, name text);'
+  gcloud spanner databases execute-sql "${GOOGLE_CLOUD_DATABASE_WITH_VERSION}" --instance="${GOOGLE_CLOUD_INSTANCE}" --sql="INSERT INTO users (id, age, name) VALUES (1, 1, 'John'), (2, 20, 'Joe'), (3, 23, 'Jack');"
```
Comment (libingye816) [inline .ci/run-with-credentials.sh]: LGTM.

Can you rewrite L73 to retry for at least two times? Though it very occasionally fails, we'd better improve this.
```
n=0
until [ "$n" -ge 2 ]
do
   command && break  # substitute your command here
   n=$((n+1)) 
   sleep 3
done
```

**Review by libingye816 — APPROVED** [review]

File: .ci/run-with-credentials.sh (hunk 1 above)
Comment (Vizerai) [reply .ci/run-with-credentials.sh]: Done.

**Pull request merged**